File: KITE-Example-Test/src/main/java/org/webrtc/kite/example/checks/GoogleFirstResultCheck.java
package org.webrtc.kite.example.checks;

import io.cosmosoftware.kite.exception.KiteTestException;
import io.cosmosoftware.kite.interfaces.Runner;
import io.cosmosoftware.kite.report.Status;
import io.cosmosoftware.kite.steps.TestCheck;
import org.webrtc.kite.example.pages.GoogleResultPage;

import static io.cosmosoftware.kite.util.ReportUtils.saveScreenshotPNG;

public class GoogleFirstResultCheck extends TestCheck {
  private final String EXPECTED_RESULT = "CoSMo Software | WebRTC Technology & Implementation";
  private final GoogleResultPage resultPage;
  
  public GoogleFirstResultCheck(Runner runner) {
    super(runner);
    this.resultPage = new GoogleResultPage(runner);
  }
  
  @Override
  protected void step() throws KiteTestException {
    resultPage.openFirstResult();
    String found = resultPage.getTitle().trim();
    if (!found.equalsIgnoreCase(EXPECTED_RESULT)) {
      throw new KiteTestException("The title of the first Google result was not correct: \n" +
        "Expected: " + EXPECTED_RESULT + " but found " + found, Status.FAILED);
    }
    reporter.screenshotAttachment(report, saveScreenshotPNG(webDriver));
  }
  
  @Override
  public String stepDescription() {
    return "Open first result on Google result page and verify the page title";
  }
}

File: KITE-Example-Test/js/checks/GoogleFirstResultCheck.js
const {TestUtils, TestStep, Status, KiteTestError} = require('kite-common');
/**
 * Class: GoogleResultPage
 * Extends: TestStep
 * Description:
 */
class GoogleResultPage extends TestStep {
  constructor(kiteBaseTest) {
    super();
    this.driver = kiteBaseTest.driver;
    this.timeout = kiteBaseTest.timeout;
    this.expectedResult = "CoSMo Software | WebRTC Technology & Implementation";
    this.page = kiteBaseTest.page;

    // Test reporter if you want to add attachment(s)
    this.testReporter = kiteBaseTest.reporter;
  }

  stepDescription() {
    return "Open first result on Google result page and verify the page title";
  }

  async step() {
    await this.page.openFirstResult(this.driver);
    let found = await this.page.getTitle(this.driver);
    if (found != this.expectedResult) {
      throw new KiteTestError(Status.FAILED, "The title of the first Google result was not correct: \n" +
      "Expected: " + this.expectedResult + " but found " + found);
    }

    // Screenshot
    let screenshot = await TestUtils.takeScreenshot(this.driver);
    this.testReporter.screenshotAttachment(this.report, "Screenshot step", screenshot); 
  }
}

module.exports = GoogleResultPage;